### PR TITLE
fix: [SRM-14020]: applied the new flow for downtime history and list view and addressed older review comments

### PR DESCRIPTION
**Pull request by karansaraswat19**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by karansaraswat19** [comment]
retrigger jest

File: src/modules/85-cv/pages/slos/CVSLODetailsPage/DetailsPanel/DetailsPanel.tsx [hunk 1]
```diff
@@ -72,8 +72,14 @@ const DetailsPanel: React.FC<DetailsPanelProps> = ({
     lazy: true
   })
 
-  const downtimeStartTime = chartTimeRange?.startTime || sloDashboardWidget?.currentPeriodStartTime || 0
-  const downtimeEndTime = chartTimeRange?.endTime || sloDashboardWidget?.currentPeriodEndTime || 0
+  const downtimeStartTime = useMemo(
+    () => chartTimeRange?.startTime || sloDashboardWidget?.currentPeriodStartTime || 0,
+    [chartTimeRange?.startTime, sloDashboardWidget?.currentPeriodStartTime]
+  )
+  const downtimeEndTime = useMemo(
```
Comment (praneshg239) [inline src/modules/85-cv/pages/slos/CVSLODetailsPage/DetailsPanel/DetailsPanel.tsx:79]: No need to use `useMemo` for simple assignments. Using more of memoization is an overhead as well. No need to change here now, but in future please use memoization only for those values which comes out of a complex calculation.

File: src/modules/85-cv/pages/slos/SLOCard/SLOTargetChartWithChangeTimeline.tsx [hunk 2]
```diff
@@ -81,14 +81,18 @@ const SLOTargetChartWithChangeTimeline: React.FC<SLOTargetChartWithChangeTimelin
...
-  const downtimeSeries = downtimeInstanceUnavailability?.map(item => {
-    return {
-      x: defaultTo(item?.startTime, 0) * 1000,
-      y: maxXLimit * 0.95,
-      startTime: defaultTo(item?.startTime, 0) * 1000,
-      endTime: defaultTo(item?.endTime, 0) * 1000
-    }
-  })
+  const downtimeSeries = useMemo(
+    () =>
```
Comment (praneshg239) [inline src/modules/85-cv/pages/slos/SLOCard/SLOTargetChartWithChangeTimeline.tsx:85]: This is a correct use case 👍 , where you are looping something which could be costly when the number of items in the array increases.

File: src/modules/85-cv/pages/slos/SLODowntimePage/components/DowntimeHistory/DowntimeHistory.tsx [hunk 3]
```diff
@@ -159,11 +168,11 @@ const DowntimeHistory = (): JSX.Element => {
 
   return (
     <Container margin={'xlarge'} padding={{ left: 'small', right: 'small' }}>
-      <DowntimeFilters />
+      {(!!content?.length || (content?.length === 0 && appliedSearchAndFilter)) && <DowntimeFilters />}
```
Comment (praneshg239) [inline src/modules/85-cv/pages/slos/SLODowntimePage/components/DowntimeHistory/DowntimeHistory.tsx:171]: please move this logic to a util function

File: src/modules/85-cv/pages/slos/SLODowntimePage/SLODowntimePage.tsx [hunk 4]
```diff
@@ -144,7 +171,16 @@ export const SLODowntimePage = (): JSX.Element => {
         }}
       >
         <Page.Body className={css.pageBody} loading={!appliedSearchAndFilter && downtimeDataLoading}>
-          {downtimeData?.data?.content?.length || error || appliedSearchAndFilter ? (
+          {!downtimeData?.data?.content?.length &&
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/SLODowntimePage/SLODowntimePage.tsx:174]: These conditions can be moved to a function.

File: src/modules/85-cv/pages/slos/SLODowntimePage/components/DowntimeList/DowntimeList.tsx [hunk 5]
```diff
@@ -336,7 +355,7 @@ const DowntimeList = ({
     <>
       <Page.SubHeader>{getAddDowntimeButton(handleCreateButton, getString)}</Page.SubHeader>
       <Container margin={'xlarge'} padding={{ left: 'small', right: 'small' }}>
-        <DowntimeFilters />
+        {(!!content?.length || (content?.length === 0 && appliedSearchAndFilter)) && <DowntimeFilters />}
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/SLODowntimePage/components/DowntimeList/DowntimeList.tsx:358]: Please move these conditions to a function.

File: src/modules/85-cv/pages/slos/CVSLODetailsPage/DetailsPanel/DetailsPanel.tsx (hunk 1 above)
Comment (karansaraswat19) [reply src/modules/85-cv/pages/slos/CVSLODetailsPage/DetailsPanel/DetailsPanel.tsx:79]: @praneshg239 This was @harshilbits  comment in the last PR. I can remove it if you say.

**Review by praneshg239 — APPROVED** [review]

File: src/modules/85-cv/pages/slos/SLODowntimePage/SLODowntimePage.tsx [hunk 6]
```diff
@@ -91,6 +91,12 @@ export const SLODowntimePage = (): JSX.Element => {
     error: downtimeHistoryError
   } = useGetHistory({ ...pathParams, queryParams, lazy: true })
 
+  const shouldRenderNoDataCard = useMemo(
+    () =>
+      !downtimeData?.data?.content?.length && !downtimeHistoryData?.data?.content?.length && !appliedSearchAndFilter,
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/SLODowntimePage/SLODowntimePage.tsx:96]: This logic can be moved to a util function.

**Review by praneshg239 — APPROVED** [review]

**Review by harshilbits — APPROVED** [review]

**Pull request merged**